### PR TITLE
image recipes for prodigy using tensorflow object detection api

### PR DESCRIPTION
**Pull request by Abhijit-2592**
The following tasks will be addressed by this pull request
- [x]  model-in-the-loop bounding box annotations with  pre-trained models from the Tensorflow object detection api model zoo: using the  frozen graphs
- [x]  A tensorflow serving hook for the above (this uses SavedModel instead of frozen graph)
- [x]  A speed and accuracy trade-off study for different meta and backbone architectures
- [x]  Support model training in the loop

All the recipes will support arbitrary image dimensions (H,W,3) because, the image resizing operation is baked into the tensorflow graph

### CODE REVIEW

**Comment by Abhijit-2592** [comment]
@ines is it possible to view the confidence score also in the UI?

**span** object's contents

```python
{
            "score": prob,
            "label": name,
            "label_id": int(class_id),
            "points": rel_points,
            "hidden": hidden,
}
```
The score doesn't seem to render in the UI

**Comment by honnibal** [comment]
@Abhijit-2592 You can add a `meta` dict with information that should show up at the bottom of the card. So you a key like `"meta": {"score": 0.6}` in the task object.

**Comment by Abhijit-2592** [comment]
@honnibal  I have added the [speed and accuracy study](https://github.com/Abhijit-2592/prodigy-recipes/blob/master/image/tf_odapi/docs/model_performance.md). Let me know your thoughts on the study.

**Comment by Abhijit-2592** [comment]
@honnibal 
Surprisingly, the **Intel® MKL-DNN** optimized Tensorflow is slower compared to the **unoptimized pip** version of Tensorflow for most of the **SSD** architectures. But, the **Faster-RCNN** and  **R-FCN** architectures are faster than their **unoptimized pip** version as expected. I am unable to reason out for the **SSD**. Any idea on  what is causing this?

**Comment by Abhijit-2592** [comment]
@honnibal 
Surprisingly, the Intel® MKL-DNN optimized Tensorflow is slower compared to the unoptimized pip version of Tensorflow for most of the SSD architectures. But, the Faster-RCNN and R-FCN architectures are faster than their unoptimized pip version as expected. I am unable to reason out for the SSD. Any idea on what is causing this?

**Comment by Abhijit-2592** [comment]
@honnibal , The **model training in the loop** recipe is also ready.

**Comment by Abhijit-2592** [comment]
[Forked the models repo](https://github.com/Abhijit-2592/models) so, people can fall back to the fork if the original changes a lot in future.

Additionally, the forked repo contains:

- A simple [script](https://github.com/Abhijit-2592/models/blob/master/research/setup_odapi.py) to setup the object detection API.
- The [exporter.py](https://github.com/Abhijit-2592/models/blob/master/research/object_detection/exporter.py) script is edited with the script given in the misc folder.

**Comment by honnibal** [comment]
Really interesting results for the time vs accuracy trade-off! So if I'm reading this right, the two winning models are:

* `ssd_resnet50_v1_fpn_shared_box_predictor_coco`: 830ms, 35 MAP
* `faster_rcnn_inception_v2_coco`: 355ms, 28 MAP

Is the time given prediction per frame? Even if so, 830ms per image might just be quick enough to keep up with annotation. If not, I'd say 355ms per image is likely good enough.

**Comment by Abhijit-2592** [comment]
@honnibal Yes, the prediction time is per image/frame. The mAP values are taken as such from the original [model zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md). As you said, those two models have a really good speed vs accuracy trade-off but, the CPU time you mentioned is for the optimized version of Tensorflow (Intel MKL-DNN optimizations).

Also, I did a few experiments with `faster_rcnn_inception_v2_coco` including training the model on fly which I shared on [twitter](https://twitter.com/balaji_abhijit/status/1146445022042050560). Apart from the initial waiting period, I didn't see much lags while annotating.
